tools/internal_ci/macos/grpc_build_artifacts.sh

```diff
@@ -23,7 +23,7 @@ cd $(dirname $0)/../../..
 
 export PREPARE_BUILD_INSTALL_DEPS_CSHARP=true
 export PREPARE_BUILD_INSTALL_DEPS_PYTHON=true
-export PREPARE_BUILD_INSTALL_DEPS_RUBY=true
```

Comment:
yes, you're right, I checked and there are no ruby artifact being built on mac in v1.46.x, so the ruby dependencies can be removed from this script. Changes to this file are a better version of https://github.com/grpc/grpc/pull/32026/files.